knowledge_engine/tests/storage/test_milvus_backend.py

```
@@ -281,6 +283,154 @@ def test_retrieve_vector_mode(self, mock_milvus_vs):
+        mock_milvus_vs.return_value = mock_store
+
+        mock_embed_model = MagicMock()
+
+        backend = MilvusBackend(
+            {
+                "url": "http://localhost:19530/default",
+                "indexStrategy": {"mode": "per_dataset", "prefix": "test"},
+            }
+        )
+
+        with pytest.raises(ValueError, match="RetrievalScope.document_ids"):
```

Ruff (0.15.20)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)